packages/pricing/src/services/price-set-money-amount-rules.ts

```diff
@@ -0,0 +1,124 @@
+}
+
+export default class PriceSetMoneyAmountRulesService<
+  TEntity extends PriceSetMoneyAmountRules = PriceSetMoneyAmountRules
+> {
+  protected readonly priceSetMoneyAmountRulesRepository_: DAL.RepositoryService
+
+  constructor({ priceSetMoneyAmountRulesRepository }: InjectedDependencies) {
+    this.priceSetMoneyAmountRulesRepository_ =
+      priceSetMoneyAmountRulesRepository
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async retrieve(
+    priceSetMoneyAmountRulesId: string,
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity> {
+    return (await retrieveEntity<
+      PriceSetMoneyAmountRules,
+      PricingTypes.PriceSetMoneyAmountRulesDTO
+    >({
+      id: priceSetMoneyAmountRulesId,
+      identifierColumn: "id",
+      entityName: PriceSetMoneyAmountRules.name,
+      repository: this.priceSetMoneyAmountRulesRepository_,
+      config,
+      sharedContext,
+    })) as TEntity
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async list(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity[]> {
+    return (await this.priceSetMoneyAmountRulesRepository_.find(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as TEntity[]
+  }
+
+  @InjectManager("priceSetMoneyAmountRulesRepository_")
+  async listAndCount(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<[TEntity[], number]> {
+    return (await this.priceSetMoneyAmountRulesRepository_.findAndCount(
+      this.buildQueryForList(filters, config),
+      sharedContext
+    )) as [TEntity[], number]
+  }
+
+  private buildQueryForList(
+    filters: PricingTypes.FilterablePriceSetMoneyAmountRulesProps = {},
+    config: FindConfig<PricingTypes.PriceSetMoneyAmountRulesDTO> = {}
+  ) {
+    const queryOptions = ModulesSdkUtils.buildQuery<PriceSetMoneyAmountRules>(
+      filters,
+      config
+    )
+
+    if (filters.id) {
+      queryOptions.where["id"] = { $in: filters.id }
+    }
+
+    return queryOptions
+  }
+
+  @InjectTransactionManager(
+    shouldForceTransaction,
+    "priceSetMoneyAmountRulesRepository_"
+  )
+  async create(
+    data: PricingTypes.CreatePriceSetMoneyAmountRulesDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TEntity[]> {
+    return (await this.priceSetMoneyAmountRulesRepository_.create(
+      data,
+      sharedContext
+    )) as TEntity[]
+  }
+
+  @InjectTransactionManager(
+    shouldForceTransaction,
+    "priceSetMoneyAmountRulesRepository_"
+  )
+  async update(
+    data: PricingTypes.UpdatePriceSetMoneyAmountRulesDTO[],
```

Comment:
suggestion(non blocking): As we discussed sometimes ago, we should maybe think about having the internal services types in the package itself, and the public api (module service) types in the types package. It looks like duplication but if we need to update the internal types it might defer from the one exposed publically. We can discuss that during a call.

cc @olivermrbl @carlos-r-l-rodrigues